tx_json.go

```diff
@@ -0,0 +1,457 @@
+			case '1':
+				params.WriteByte(byte(val)) // Write to buffer
+			case '2':
+				binary.LittleEndian.PutUint16(intBuf[:2], uint16(val)) // Write to buffer
+				params.Write(intBuf[:2])                               // Write to buffer
+			case '4':
+				binary.LittleEndian.PutUint32(intBuf[:4], uint32(val)) // Write to buffer
+				params.Write(intBuf[:4])                               // Write to buffer
+			case '8':
+				binary.LittleEndian.PutUint64(intBuf[:8], uint64(val)) // Write to buffer
+				params.Write(intBuf[:8])                               // Write to buffer
+			}
+		case 'H':
+			buf, err := hex.DecodeString(payload[1:]) // Decode hex string
+			if err != nil {                           // Check for errors
+				return [8]byte{}, nil, err // Return found error
+			}
+
+			params.Write(buf) // Write to params
+		default:
+			return [8]byte{}, nil, nil // No params
+		}
+	}
+
+	binary.LittleEndian.PutUint32(intBuf[:4], uint32(len(params.Bytes()))) // Write length of function parameters to buffer
+
+	return intBuf, params.Bytes(), nil // Return params
+}
+
+// parseContractCode gets the code of a particular payload's corresponding
+// contract.
+func parseContractCode(json *fastjson.Value) ([]byte, error) {
+	if !json.Exists("contract_code") { // Check no value
+		return nil, ErrNilField // Return nil field error
+	}
+
+	code, err := ioutil.ReadFile(string(json.GetStringBytes("contract_code"))) // Read contract code
+	if err != nil {                                                            // Check for errors
+		return nil, err // Return found error
+	}
+
+	return code, nil // Return contract code
+}
+
+// parseRecipient gets the recipient of the transaction from the JSON map,
+// and performs address length safety checks immediately after.
+func parseRecipient(json *fastjson.Value) ([]byte, error) {
+	if !json.Exists("recipient") { // Check no value
+		return nil, ErrNilField // Return nil field error
+	}
+
+	recipient, err := hex.DecodeString(string(json.GetStringBytes("recipient"))) // Decode recipient hex string
+	if err != nil {                                                              // Check for errors
+		return nil, err // Return found error
+	}
+
+	if len(recipient) != SizeAccountID {
+		return nil, err // Return invalid recipient ID error
+	}
+
+	var recipientID AccountID       // Initialize ID buffer
+	copy(recipientID[:], recipient) // Copy address to buffer
+
+	return recipient, nil // Return recipient
+}
+
+// parseJSONBytes parses a given raw JSON input, and converts such an input
+// to a fastjson value pointer.
+func parseJSONBytes(data []byte) (*fastjson.Value, error) {
+	var p fastjson.Parser // Initialize parser
+
+	v, err := p.Parse(string(data)) // Parse json
+	if err != nil {                 // Check for errors
+		return nil, err // Return found error
+	}
+
+	return v, nil // Return parsed
+}
+
+// getValidTags gets a populated map of valid tags.
+func getValidTags() map[string]bool {
```

Comment:
I would use `stringer` https://godoc.org/golang.org/x/tools/cmd/stringer and generate the string names for each individual operation type. The constant array of strings could go in the `sys` pkg under `const.go`, or otherwise in a separate file called `op.go`.